modules/sql/src/main/scala/EnumDef.scala

```diff
@@ -80,9 +80,9 @@ object EnumDef {
     implicit def caseOptionMagnitudeValue[S <: Symbol] = at[(S, Option[MagnitudeValue])] { case (s, _) => s"  val ${s.name}: Option[MagnitudeValue]" }
     implicit def caseGnirsPixelScale[S <: Symbol] = at[(S, GnirsPixelScale)] { case (s, _) => s"  val ${s.name}: GnirsPixelScale" }
 
-    implicit def caseEnumRef[T <: Symbol: ValueOf, S <: Symbol]       = at[(S, EnumRef[T])        ] { case (s, _) => s"  val ${s.name}: ${valueOf[T].name}" }
-    implicit def caseLazyEnumRef[T <: Symbol: ValueOf, S <: Symbol]   = at[(S, LazyEnumRef[T])    ] { case (s, _) => s"  val ${s.name}: Eval[${valueOf[T].name}]" }
-    implicit def caseOptionEnumRef[T <: Symbol: ValueOf, S <: Symbol] = at[(S, Option[EnumRef[T]])] { case (s, _) => s"  val ${s.name}: Option[${valueOf[T].name}]" }
+    implicit def caseEnumRef[T <: Symbol, S <: Symbol](implicit w: Witness.Aux[T])       = at[(S, EnumRef[T])        ] { case (s, _) => s"  val ${s.name}: ${w.value.name}" }
```

Comment:
How does `Witness` differ from `ValueOf`?

Reply:
`Witness` kind of serves double duty. It lets you express both singleton types and singleton values (i.e., values that have an ascribed singleton type).

So, with literal types (which we will get back in 2.13) you can say

```scala
val op: Option[42] = Some(42)
```

You can't do that in Lightbend 2.12 … `42` can't appear in type position, and the value `42` has type `Int`. But you can do it with `Witness`.

```scala
val op: Option[Witness.`42`.T] = Some(Witness(42).value) 
val op: Option[Witness.`42`.T] = Some(42.narrow) // syntax 
```

The `42` in backticks is a trick. It's a method call (!) on `object Witness` (which extends `Dynamic`). The `selectDynamic` method is implemented with a macro that parses and typechecks the dynamic method name (!) and constructs the singleton type, which you can't express in Scala but *can* express in terms of the underlying AST.

`Witness.Aux[A]` is always available for singleton types `T` so you can write `valueOf` in terms of it.

```scala
@ def valueOf[A](implicit ev: Witness.Aux[A]): A = ev.value 
defined function valueOf

@ valueOf[Witness.`42`.T] 
res11: Int = 42
```